### PR TITLE
[Backport release/3.2.x] fix(bazel): fix intel detection

### PR DESCRIPTION
**Pull request by mashapedeployment**
Backport 50e8ba2df5c6134407fef034b4b170316cf239ab from #10322.

### CODE REVIEW

**Comment by curiositycasualty** [comment]
> Didn't this break something else in the pipeline? I don't remember the specifics but at some point this week there was an issue with this changeset

> I don't want to merge that before the 3.2.1 release in case it affects the release

~ @tyler-ball

**Comment by fffonion** [comment]
I have amend a fix from #10378, let's wait for 10378 to merge and verify master is good before merge this.

**Review by fffonion — APPROVED** [review]

**Pull request merged**
